steward/src/cellars/aave_v2_stablecoin.rs

```diff
@@ -1,77 +1,143 @@
+                    Ok(addr) => Ok(addr),
+                    Err(_) => Err(addr),
+                })
+                .collect();
+
+            validate_route(results.clone())?;
+
+            let route = results
+                .iter()
+                .map(|r| r.unwrap())
+                .collect::<Vec<H160>>()
+                .try_into()
+                .expect("failed to convert 'route' addresses to array");
+
+            let swap_params = params
+                .swap_params
+                .iter()
+                .map(|sp| {
+                    let out: [U256; 3] =
+                        [sp.in_index.into(), sp.out_index.into(), sp.swap_type.into()];
+                    out
+                })
+                .collect::<Vec<[U256; 3]>>()
+                .try_into()
+                .expect("failed to convert 'swap_params' vec to array");
+
+            let call = RebalanceCall {
+                route,
+                swap_params,
                 min_assets_out: params.min_assets_out.into(),
             };
-            AaveV2StablecoinCellarCalls::ReinvestWithAmount(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::Rebalance(call).encode())
         }
         Reinvest(params) => {
             let call = ReinvestCall {
                 min_assets_out: params.min_assets_out.into(),
             };
-            AaveV2StablecoinCellarCalls::Reinvest(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::Reinvest(call).encode())
         }
-        ClaimAndUnstakeAmount(params) => {
-            let call = ClaimAndUnstakeWithAmountCall {
-                amount: params.amount.into(),
+        SetDepositLimit(params) => {
+            let call = SetDepositLimitCall {
+                limit: params.limit.into(),
             };
-            AaveV2StablecoinCellarCalls::ClaimAndUnstakeWithAmount(call).encode()
-        }
-        ClaimAndUnstake(_) => {
-            let call = ClaimAndUnstakeCall {};
-            AaveV2StablecoinCellarCalls::ClaimAndUnstake(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::SetDepositLimit(call).encode())
         }
-        Rebalance(params) => {
-            let call = RebalanceCall {
-                new_lending_token: params
-                    .address
-                    .parse::<H160>()
-                    .expect("failed to parse token address"),
-                min_new_lending_token_amount: params.min_new_lending_token_amount.into(),
+        SetLiquidityLimit(params) => {
+            let call = SetLiquidityLimitCall {
+                limit: params.limit.into(),
             };
-            AaveV2StablecoinCellarCalls::Rebalance(call).encode()
+            Ok(AaveV2StablecoinCellarCalls::SetLiquidityLimit(call).encode())
         }
-        AccruePlatformFees(_) => {
-            let call = AccruePlatformFeesCall {};
-            AaveV2StablecoinCellarCalls::AccruePlatformFees(call).encode()
+        Sweep(params) => {
```

Comment:
I think we should leave sweep out of the SP functions for now. We can add it back later if we think SPs should be able to call it.